src/TaskSerializer/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { Task } from '../Task';
+
+type Writeable<T> = { -readonly [P in keyof T]: T[P] };
+
+/**
+ * A subset of fields of {@link Task} that can be parsed from the textual
+ * description of that Task.
+ *
+ * All fields are writeable for convenience.
+ */
+export type TaskDetails = Writeable<
+    Pick<
+        Task,
+        | 'description'
+        | 'priority'
+        | 'startDate'
+        | 'createdDate'
+        | 'scheduledDate'
+        | 'dueDate'
+        | 'doneDate'
+        | 'recurrence'
+        | 'tags'
+    >
+>;
+
+/**
+ * An abstraction that manages how a {@link Task} is read from and written
+ * to a file.
+ *
+ * A {@link TaskSerializer} is only responsible for the single line of text that follows
+ * after the checkbox:
+ *
+ *        - [ ] This is a task description
+ *              ~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
```

Comment:
I think it might be worth listing other behaviours it is not responsible for, such as the setting of the scheduled date from the file name.

I think it _is_ responsible for trimming white-space from tags, and uses a helper to do that. Unsure if that is worth noting here though.